src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -103,7 +102,8 @@
     private final Interpolator interpolator;
     private final ClientUtil clientUtil;
     private int maxEntitiesForPreview;
-    private final Gson gson;
+    private int pageSize;
```

Comment:
Add volatile ?

Reply:
added

Reply:
No, I always use volatile before fields which shared by multiple threads for system settings.